Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -261,6 +261,35 @@ namespace Babylon
             DoForHandleTypes(nonDynamic, dynamic);
         }
 
+        template<typename sourceType, bool normalized> void PromoteToFloats(uint32_t numElements, uint32_t byteStride)
+        {
+            const size_t count = m_bytes.size() / byteStride;
+            const size_t destinationSize = count * numElements * sizeof(float);
+            const float maxRange = static_cast<float>(std::numeric_limits<sourceType>::max());
+            if (destinationSize != m_bytes.size()) // ensure both vectors have different size
+            {
+                std::vector<uint8_t> bytes(destinationSize);
+                float* destination = (float*)bytes.data();
```

Comment:
nit: missed another one
```suggestion
                float* destination = reinterpret_cast<float*>(bytes.data());
```